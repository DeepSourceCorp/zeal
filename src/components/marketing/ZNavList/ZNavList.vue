<template>
  <ul :class="[SLOT_PADDING]">
    <li v-if="$slots.heading" class="px-4 pt-2 pb-4 uppercase text-sm text-vanilla-400">
      <slot name="heading"></slot>
    </li>
    <span :class="[ARRANGEMENT[arrange].classes]">
      <slot></slot>
    </span>
  </ul>
</template>

<script lang="ts">
import Vue from 'vue'

const SLOT_PADDING = 'p-2'

const ARRANGEMENT = {
  vertical: {
    text: 'vertical',
    classes: ''
  },
  horizontal: {
    text: 'horizontal',
    classes: 'flex flex-wrap'
  }
}

export default Vue.extend({
  name: 'ZNavList',
  data() {
    return {
      ARRANGEMENT,
      SLOT_PADDING
    }
  },
  props: {
    arrange: {
      type: String,
      default: ARRANGEMENT.vertical.text,
      validator: (arrange) => Object.keys(ARRANGEMENT).includes(arrange)
    }
  }
})
</script>
