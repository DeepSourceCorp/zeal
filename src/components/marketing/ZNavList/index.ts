import ZNavList from './ZNavList.vue'

export default ZNavList
