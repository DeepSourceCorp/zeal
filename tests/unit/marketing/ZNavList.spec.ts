import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZNavList from '../../../src/components/marketing/ZNavList'

describe('ZNavList component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZNavList, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZNavList when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
})
