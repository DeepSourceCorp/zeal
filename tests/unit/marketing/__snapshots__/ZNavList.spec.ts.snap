// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZNavList component renders a ZNavList when the component is mounted 1`] = `
<ul class="p-2">
  <!----> <span class=""></span>
</ul>
`;
